src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -73,28 +74,66 @@ public class Arguments : UnsupervisedLearnerInputBaseWithWeight
         private readonly int _oversampling;
         private readonly bool _center;
         private readonly int _seed;
+        private readonly string _featureColumn;
 
         public override PredictionKind PredictionKind => PredictionKind.AnomalyDetection;
 
         // The training performs two passes, only. Probably not worth caching.
         private static readonly TrainerInfo _info = new TrainerInfo(caching: false);
         public override TrainerInfo Info => _info;
 
-        public RandomizedPcaTrainer(IHostEnvironment env, Arguments args)
-            : base(env, LoadNameValue)
+        /// <summary>
+        /// Initializes a new instance of <see cref="RandomizedPcaTrainer"/>.
+        /// </summary>
+        /// <param name="env">The local instance of the <see cref="IHostEnvironment"/>.</param>
+        /// <param name="featureColumn">The name of the feature column.</param>
+        /// <param name="weightColumn">The name of the weight column.</param>
+        /// <param name="rank">The number of components in the PCA.</param>
+        /// <param name="oversampling">Oversampling parameter for randomized PCA training.</param>
+        /// <param name="center">If enabled, data is centered to be zero mean.</param>
+        /// <param name="seed">The seed for random number generation.</param>
```

Comment:
>seed [](start = 25, length = 4)

why do we use a separate seed here?

Reply:
it is part of the args, therefore i moved it here; rather than doing the delegate on advancedArgs. 

---
In reply to: [220353522](https://github.com/dotnet/machinelearning/pull/996#discussion_r220353522) [](ancestors = 220353522)